cupy/polynomial/polyutils.py

```diff
@@ -108,3 +108,24 @@ def trimcoef(c, tol=0):
     if ind == 0:
         return cupy.zeros_like(c[:1])
     return c[: ind]
+
+
+def _add(c1, c2):
+    """Helper function used in ``<type>add`` functions.
+    """
+    if c1.size > c2.size:
+        c1[:c2.size] += c2
+        return c1
```

Comment:
The input array should not be overwritten.

Reply:
The inputs won't be overwritten because of `_wraps_chebroutine` wrapper (which is called before these internal operations are called). I think all polynomial series (power, chebyshev...) will need their own preprocessing before calling `_add`/`_sub` so probably no overwriting will occur when reusing them for other series as well, right?

Reply:
So, do you mean that the wrapper copies the inputs before calling this function? If so, in which line of your code?